packages/@react-aria/interactions/src/useFocus.ts

```diff
@@ -33,49 +33,42 @@ interface FocusResult {
  * Handles focus events for the immediate target.
  * Focus events on child elements will be ignored.
  */
-export function useFocus(props: FocusProps): FocusResult {
-  let onBlur: FocusProps['onBlur'];
-  if (!props.isDisabled && (props.onBlur || props.onFocusChange)) {
-    onBlur = (e: FocusEvent) => {
-      if (e.target === e.currentTarget) {
-        if (props.onBlur) {
-          props.onBlur(e);
-        }
-
-        if (props.onFocusChange) {
-          props.onFocusChange(false);
-        }
+export function useFocus({isDisabled, onFocus: onFocusProp, onBlur: onBlurProp, onFocusChange}: FocusProps): FocusResult {
+  const onBlur: FocusProps['onBlur'] = useCallback((e: FocusEvent) => {
+    if (e.target === e.currentTarget) {
+      if (onBlurProp) {
+        onBlurProp(e);
+      }
 
-        return true;
+      if (onFocusChange) {
+        onFocusChange(false);
       }
-    };
-  } else {
-    onBlur = null;
-  }
 
-  let onSyntheticFocus = useSyntheticBlurEvent(onBlur);
+      return true;
+    }
+  }, [onBlurProp, onFocusChange]);
 
-  let onFocus: FocusProps['onFocus'];
-  if (!props.isDisabled && (props.onFocus || props.onFocusChange || props.onBlur)) {
-    onFocus = (e: FocusEvent) => {
-      if (e.target === e.currentTarget) {
-        if (props.onFocus) {
-          props.onFocus(e);
-        }
 
-        if (props.onFocusChange) {
-          props.onFocusChange(true);
-        }
+  const onSyntheticFocus = useSyntheticBlurEvent(onBlur);
 
-        onSyntheticFocus(e);
+  const onFocus: FocusProps['onFocus'] = useCallback((e: FocusEvent) => {
+    if (e.target === e.currentTarget) {
+      if (onFocusProp) {
+        onFocusProp(e);
       }
-    };
-  }
+
+      if (onFocusChange) {
+        onFocusChange(true);
+      }
+
+      onSyntheticFocus(e);
+    }
+  }, [onFocusChange, onFocusProp, onSyntheticFocus]);
 
   return {
     focusProps: {
-      onFocus,
-      onBlur
+      onFocus: (!isDisabled && (onFocusProp || onFocusChange || onBlurProp)) ? onFocus : undefined,
+      onBlur: (!isDisabled && (onBlurProp || onFocusChange)) ? onBlur : null
```

Comment:
out of curiosity, why undefined for one but null for the other?

Reply:
😬 I kept the logic that was there. not really sure why it was like that

Reply:
onBlur null: https://github.com/adobe/react-spectrum/pull/3045/files#diff-70aae225f0b9efb36943afd209d8423f070cc51965757699da4ab865b337ee3dL53

onFocus undefined: https://github.com/adobe/react-spectrum/pull/3045/files#diff-70aae225f0b9efb36943afd209d8423f070cc51965757699da4ab865b337ee3dL58

Reply:
ah, thanks, didn't notice it